tsconfig.json

```diff
@@ -2,12 +2,10 @@
 	"extends": "@sindresorhus/tsconfig",
 	"compilerOptions": {
 		"outDir": "dist",
-		"target": "es2020", // Node.js 14
-		"lib": [
-			"es2020"
+	    "moduleResolution": "Node",
```

Comment:
This should not be here

Reply:
Atm got 12.3.1 has `moduleResolution=Node` inherited from `@sindresorhus/tsconfig 2.0.0`

Switching to `@sindresorhus/tsconfig 3.0.1` sets `moduleResolution` to `Node16` which breaks type checking when building got (due to issues with how some packages interact with `moduleReolution=Node16`, see debugging attempts in https://github.com/sindresorhus/got/issues/2051#issuecomment-1214196483 )

Overriding moduleResolution to Node here allows to upgrade tsconfig without breaking the package. 
It should ultimately be removed when the resolution with the problematic packages is okay. 

I added a comment in the file, does that sound reasonable?